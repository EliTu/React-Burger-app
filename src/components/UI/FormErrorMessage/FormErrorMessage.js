import React from 'react';
import styles from './FormErrorMessage.module.css';
import PropTypes from 'prop-types';

const FormErrorMessage = ({ errorType }) => {
	// CSS Modules styles:
	const { FormErrorMessage } = styles;

	let errorMessage;
	switch (errorType) {
		case 'noMatch':
			errorMessage = (
				<p>
					The passwords you entered did not match! please try again.
				</p>
			);
			break;
		case 'emptyFields':
			errorMessage = (
				<p>
					Please properly fill out all the required fields before
					submitting.
				</p>
			);
			break;
		default:
			errorMessage = '';
			break;
	}

	return <div className={FormErrorMessage}>{errorMessage}</div>;
};

FormErrorMessage.propTypes = {
	errorType: PropTypes.string,
};

export default FormErrorMessage;
