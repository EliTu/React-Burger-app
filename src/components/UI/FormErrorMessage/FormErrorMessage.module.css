@import '../../../variables.css';

.FormErrorMessage p {
    display: inline-block;
    font-size: 0.9rem;
    color: var(--invalid-red-dark);
}