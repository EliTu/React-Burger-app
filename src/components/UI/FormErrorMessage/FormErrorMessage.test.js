import React from 'react';
import FormErrorMessage from './FormErrorMessage';
import { shallow } from 'enzyme';

describe('<FormErrorMessage>', () => {
	let component = shallow(<FormErrorMessage />);

	it('should render the component without errors', () => {
		expect(component).toMatchSnapshot();
		expect(component.length).toBe(1);
		expect(component.length).not.toBe(2);
	});

	it('should render a specific error message if errorType is noMatch', () => {
		component.setProps({ errorType: 'noMatch' });

		expect(component).toMatchSnapshot();
		expect(component.find('p').text()).toBe(
			'The passwords you entered did not match! please try again.'
		);
	});

	it('should render a specific error message if errorType is emptyFields', () => {
		component.setProps({ errorType: 'emptyFields' });

		expect(component).toMatchSnapshot();
		expect(component.find('p').text()).toBe(
			'Please properly fill out all the required fields before submitting.'
		);
	});
});
