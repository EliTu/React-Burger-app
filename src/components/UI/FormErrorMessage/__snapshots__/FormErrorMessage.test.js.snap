// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FormErrorMessage> should render a specific error message if errorType is emptyFields 1`] = `
<div
  className="FormErrorMessage"
>
  <p>
    Please properly fill out all the required fields before submitting.
  </p>
</div>
`;

exports[`<FormErrorMessage> should render a specific error message if errorType is noMatch 1`] = `
<div
  className="FormErrorMessage"
>
  <p>
    The passwords you entered did not match! please try again.
  </p>
</div>
`;

exports[`<FormErrorMessage> should render the component without errors 1`] = `
<div
  className="FormErrorMessage"
/>
`;
